### DISCUSSION TITLE
AttributeError: 'DeflateIO' object has no attribute 'write'

### DISCUSSION THREAD

**Discussion by shulfei** [micropython/micropython#14283] [ESP32]
Hello.
Please help with decoding a string in gzip.
I get an error `AttributeError: 'DeflateIO' object has no attribute 'write'`

What could be the problem, how to solve it?

```
import deflate
uncompressed_data = bytes(content, 'UTF-8')
stream = io.BytesIO()
with deflate.DeflateIO(stream, deflate.GZIP) as d:
    d.write(uncompressed_data)
compressed_data = stream.getvalue()
return compressed_data.decode('utf-8')
```

**Comment by sosi-deadeye** [comment]
You have to compile the firmware for yourself. A modification of mpconfigboard.h is required:
```c
#define MICROPY_PY_DEFLATE_COMPRESS (1)
```

For esp32 GENERIC:
```c
// Both of these can be set by mpconfigboard.cmake if a BOARD_VARIANT is
// specified.

#ifndef MICROPY_HW_BOARD_NAME
#define MICROPY_HW_BOARD_NAME "Generic ESP32 module"
#endif

#ifndef MICROPY_HW_MCU_NAME
#define MICROPY_HW_MCU_NAME "ESP32"
#endif

#define MICROPY_PY_DEFLATE_COMPRESS (1)
```
The file is located here: https://github.com/micropython/micropython/blob/5114f2c1ea7c05fc7ab920299967595cfc5307de/ports/esp32/boards/ESP32_GENERIC/mpconfigboard.h

**Comment by shulfei** [comment]
Thank you. I'll try.
